driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```diff
@@ -0,0 +1,148 @@
+
+    private static boolean sameParentDomain(final List<String> srvHostDomainParts, final String resolvedHostDomain) {
+        List<String> resolvedHostDomainParts = asList(resolvedHostDomain.split("\\."));
+        if (srvHostDomainParts.size() > resolvedHostDomainParts.size()) {
+            return false;
+        }
+        return resolvedHostDomainParts.subList(resolvedHostDomainParts.size() - srvHostDomainParts.size(), resolvedHostDomainParts.size())
+                .equals(srvHostDomainParts);
+    }
+
+    // A TXT record is just a string
+    // We require each to be one or more query parameters for a MongoDB connection string.
+    // Here we concatenate TXT records together with a '&' separator as required by connection strings
+    @Override
+    public String resolveAdditionalQueryParametersFromTxtRecords(final String host) {
+        String additionalQueryParameters = "";
+        InitialDirContext dirContext = createDnsDirContext();
+        try {
+            Attributes attributes = dirContext.getAttributes(host, new String[]{"TXT"});
+            Attribute attribute = attributes.get("TXT");
+            if (attribute != null) {
+                NamingEnumeration<?> txtRecordEnumeration = attribute.getAll();
+                if (txtRecordEnumeration.hasMore()) {
+                    // Remove all space characters, as the DNS resolver for TXT records inserts a space character
+                    // between each character-string in a single TXT record.  That whitespace is spurious in
+                    // this context and must be removed
+                    additionalQueryParameters = ((String) txtRecordEnumeration.next()).replaceAll("\\s", "");
+
+                    if (txtRecordEnumeration.hasMore()) {
+                        throw new MongoConfigurationException(format("Multiple TXT records found for host '%s'.  Only one is permitted",
+                                host));
+                    }
+                }
+            }
+        } catch (NamingException e) {
+            throw new MongoConfigurationException("Unable to look up TXT record for host " + host, e);
+        } finally {
+            try {
+                dirContext.close();
+            } catch (NamingException e) {
+                // ignore
+            }
+        }
+        return additionalQueryParameters;
+    }
+
+    // It's unfortunate that we take a runtime dependency on com.sun.jndi.dns.DnsContextFactory.
+    // This is not guaranteed to work on all JVMs but in practice is expected to work on most.
+    private static InitialDirContext createDnsDirContext() {
+        Hashtable<String, String> envProps = new Hashtable<String, String>();
+        envProps.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.dns.DnsContextFactory");
+        try {
+            return new InitialDirContext(envProps);
+        } catch (NamingException e) {
+            throw new MongoClientException("Unable to create JNDI context for resolving SRV records. "
```

Comment:
Suggestion make it explicit that `mongodb+srv` style connections are not supported.  Some thing like:

> Unable to support `mongodb+srv//` style connections as the  'com.sun.jndi.dns.DnsContextFactory' class is not available in this JRE. JNDI context is required for resolving SRV records.

Reply:
Done